lib/__tests__.js

```diff
@@ -1151,112 +1150,151 @@ Test('federated schema can include custom directive', (t) => {
   });
 });
 
-Test('custom makeExecutableSchema option', (t) => {
-
-  const baseComponentOptions = {
-    types: [`
-      type A {
-        value: String
+Test('pruning schema', (t) => {
+  
+  t.test('no prune', (t) => {
+    const component1 = new GraphQLComponent({
+      types: `
+        type Query {
+          property(id: ID!): Property
+        }
+  
+        type Property {
+          id: ID!
+          geo: [String]
+        }
+      `,
+      resolvers: {
+        Query: {
+          property(_, { id }) {
+            return {
+              id,
+              geo: ['lat', 'long']
+            }
+          }
+        }
       }
-      type Query {
-        a: A
-      }
-    `]
-  };
-
-  t.test('component schema uses custom makeExecutableSchema when not federated', async (st) => {
-    st.plan(2);
-    const customMakeExecutableSchema = sinon.spy();
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: customMakeExecutableSchema,
-      federation: false
     });
+  
+    const component2 = new GraphQLComponent({
+      types: `
+        type Query {
+          reviews(propertyId: ID!): [Review]
+        }
+  
+        type Review {
+          id: ID!
+          content: String
+        }
 
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
-  });
-
-  t.test('component schema uses custom makeExecutableSchema when federation is unspecified', async (st) => {
-    st.plan(2);
-    const customMakeExecutableSchema = sinon.spy();
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: customMakeExecutableSchema
+        type Property {
+          reviews: [Review]
+        }
+      `,
+      resolvers: {
+        Query: {
+          reviews() {
+            return {
+              id: 'rev-id-1',
+              content: 'content'
+            }
+          }
+        },
+      }
     });
-
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
-  });
-
-  t.test('component schema uses custom makeExecutableSchema when federation is specified but undefined', async (st) => {
-    st.plan(2);
-    const customMakeExecutableSchema = sinon.spy();
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: customMakeExecutableSchema,
-      federation: undefined
+  
+    const rootComponent = new GraphQLComponent({
+      imports: [component1, component2],
+      pruneSchema: true
     });
-
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
+  
+    const { schema } = rootComponent;
+  
+    console.log(graphql.printSchema(schema));
```

Comment:
guessing this was accidentally left while debugging?

Reply:
Right